app/serializers/optimized_project_serializer.rb

```diff
@@ -55,7 +59,7 @@ def serialize_project(project)
           repository_license: project.repository_license,
           repository_status: project.repository_status,
           stars: project.stars,
-          versions: project.versions,
+          versions: project.versions.map { |v| v.slice(*VERSION_ATTRIBUTES) },
```

Comment:
this can be something like `versions.pluck_to_hash(*VERSION_ATTRIBUTES)` potentially

Reply:
I think the obstacle is that we're preloading `project.versions` to avoid the N+1 (in ProjectStatusQuery), but using the `pluck` subverts the preload and brings back the N+1 : \

I know I've run into this problem before (N+1 plus pluck), but I can't remember if I found a solution for it : \

Reply:
wait a sec, I've got an idea

Reply:
oh that does make sense and agreed that N+1 is a worse problem than creating the models. Was just an idea